### PR TITLE
Feat: top facets category view

### PR DESCRIPTION
**Pull request by ivan-kalachikov**
### Jira-link:
https://virtocommerce.atlassian.net/browse/VCST-1349
### Artifact URL:
https://vc3prerelease.blob.core.windows.net/packages/vc-theme-b2b-vue-2.0.0-pr-1158-e2ef-e2ef352a.zip

### CODE REVIEW

File: client-app/ui-kit/components/atoms/popup-sidebar/vc-popup-sidebar.vue [hunk 1]
```diff
@@ -68,3 +68,9 @@ function onHide() {
   emit("hide");
 }
 </script>
+
+<style lang="scss" scoped>
+.vc-popup-sidebar {
+  --close-button-color: var(--color-danger-400);
```
Comment (ivan-kalachikov) [inline client-app/ui-kit/components/atoms/popup-sidebar/vc-popup-sidebar.vue:74]: To have the opportunity to change the color easily

File: client-app/shared/catalog/components/products-filters.vue [hunk 2]
```diff
@@ -103,3 +161,11 @@ function onOpenBranches(): void {
   emit("openBranches");
 }
 </script>
+
+<style scoped lang="scss">
+:deep(.facet-filter--dropdown:last-child) {
+  .vc-popover__content {
+    @apply right-0 left-auto #{!important};
```
Comment (ivan-kalachikov) [inline client-app/shared/catalog/components/products-filters.vue:168]: For aligning the last (rightmost) filter dropdown to right and avoiding it from container overflowing.

File: client-app/shared/catalog/composables/useStickyFilters.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
+import { useElementBounding } from "@vueuse/core";
```
Comment (ivan-kalachikov) [inline client-app/shared/catalog/composables/useStickyFilters.ts:1]: the logic moved from `category.vue`

File: client-app/shared/catalog/composables/useSeo.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { useSeoMeta } from "@unhead/vue";
```
Comment (ivan-kalachikov) [inline client-app/shared/catalog/composables/useSeo.ts:1]: the logic moved from `category.vue`

File: client-app/shared/catalog/components/category/category-controls.vue [hunk 5]
```diff
@@ -0,0 +1,72 @@
+<template>
```
Comment (ivan-kalachikov) [inline client-app/shared/catalog/components/category/category-controls.vue:1]: the component moved from `category.vue`

File: client-app/shared/catalog/components/category/category-products.vue [hunk 6]
```diff
@@ -0,0 +1,161 @@
+<template>
```
Comment (ivan-kalachikov) [inline client-app/shared/catalog/components/category/category-products.vue:1]: the component moved from `category.vue`

File: client-app/shared/catalog/components/category/filters-popup-sidebar.vue [hunk 7]
```diff
@@ -0,0 +1,128 @@
+<template>
```
Comment (ivan-kalachikov) [inline client-app/shared/catalog/components/category/filters-popup-sidebar.vue:1]: the component moved from `category.vue`

File: client-app/shared/catalog/composables/useSeo.ts [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+import { computed } from "vue";
+import { usePageHead } from "@/core/composables";
+import { useCategory } from "@/shared/catalog/composables/useCategory";
+
+export function useSeo() {
```
Comment (NaMax66) [inline client-app/shared/catalog/composables/useSeo.ts:6]: useCategorySeo

File: client-app/shared/catalog/components/facet-filter.vue [hunk 9]
```diff
@@ -1,43 +1,111 @@
...
+          <VcButtonSeeMoreLess v-model="isExpanded" />
+        </div>
+      </template>
+    </VcWidget>
+
+    <!-- Dropdown mode -->
+    <VcDropdownMenu v-if="mode === 'dropdown'" :offset-options="4" class="relative z-10" :max-height="maxHeight">
+      <template #trigger="{ opened }">
+        <VcButton
+          class="facet-filter__trigger border-2 border-r-4"
```
Comment (NaMax66) [inline client-app/shared/catalog/components/facet-filter.vue:48]: facet-filter__trigger also used in products-filters.vue. line 112. Lets try to refactor and use one class in one component

File: client-app/shared/catalog/components/category/filters-popup-sidebar.vue [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+      <div class="text-2xl font-bold">
+        {{ $t("common.buttons.allFilters") }}
+      </div>
+      <button type="button" class="appearance-none px-5 py-4" @click="$emit('hidePopupSidebar')">
+        <VcIcon class="!text-neutral-600" size="sm" name="x" />
+      </button>
+    </template>
+
+    <ProductsFilters
+      :show-common-filters="areHorizontalFilters"
```
Comment (NaMax66) [inline client-app/shared/catalog/components/category/filters-popup-sidebar.vue:17]: why isExistSelectedFacets but areHorizontalFilters? Maybe isHorizontalFilters ok? :)

**Review by NaMax66 — CHANGES_REQUESTED** [review]

File: client-app/shared/catalog/components/facet-filter.vue (hunk 9 above)
Comment (ivan-kalachikov) [reply client-app/shared/catalog/components/facet-filter.vue:48]: In `products-filters.vue` the class it's used in `querySelectorAll(".facet-filter__trigger")` to get access for dom elements. So I think it's okay in this case.

File: client-app/shared/catalog/components/category/filters-popup-sidebar.vue (hunk 10 above)
Comment (ivan-kalachikov) [reply client-app/shared/catalog/components/category/filters-popup-sidebar.vue]: okay, renamed :)

File: client-app/shared/catalog/composables/useSeo.ts (hunk 8 above)
Comment (ivan-kalachikov) [reply client-app/shared/catalog/composables/useSeo.ts]: renamed

**Review by NaMax66 — APPROVED** [review]

File: client-app/shared/catalog/components/facet-filter.vue [hunk 11]
```diff
@@ -125,4 +196,26 @@ const hasFade = computed(
     }
   }
 }
+
+.facet-filter--dropdown {
```
Comment (goldenmaya) [inline client-app/shared/catalog/components/facet-filter.vue:200]: forgot .facet-filter block

File: client-app/shared/catalog/components/facet-filter.vue
Comment (goldenmaya) [inline client-app/shared/catalog/components/facet-filter.vue:1]: Use BEM: block -> .facet-filter

File: client-app/shared/catalog/components/facet-filter.vue [hunk 12]
```diff
@@ -1,43 +1,111 @@
...
-
-      <div class="-me-1 space-y-3 overflow-y-auto pe-1" :style="{ maxHeight }">
-        <VcCheckbox
-          v-for="item in searchedValues"
-          :key="item.value"
-          v-model="item.selected"
+  <div :class="`facet-filter facet-filter--${mode}`">
+    <!-- Collapsable mode -->
+    <VcWidget v-if="mode === 'collapsable'" size="xs" collapsible :title="facet.label" collapsed>
+      <div :class="{ 'fade-bottom': hasFade }">
```
Comment (goldenmaya) [inline client-app/shared/catalog/components/facet-filter.vue:5]: Use BEM

File: client-app/shared/catalog/components/facet-filter.vue [hunk 13]
```diff
@@ -1,43 +1,111 @@
...
+        <VcInput
+          v-if="searchFieldVisible"
+          v-model.trim="searchKeyword"
+          size="sm"
+          class="mb-4"
```
Comment (goldenmaya) [inline client-app/shared/catalog/components/facet-filter.vue:10]: Use BEM

File: client-app/shared/catalog/components/facet-filter.vue [hunk 14]
```diff
@@ -1,43 +1,111 @@
...
+          maxlength="30"
           :disabled="loading"
-          @change="changeFacetValues"
+          :placeholder="$t('common.labels.search', [facet.label])"
+          truncate
+        />
+
+        <div class="-me-1 space-y-3 overflow-y-auto pe-1" :style="{ maxHeight }">
```
Comment (goldenmaya) [inline client-app/shared/catalog/components/facet-filter.vue:17]: Use BEM - here and below

**Review by goldenmaya — CHANGES_REQUESTED** [review]

File: client-app/shared/catalog/components/category/category-horizontal-filters.vue [hunk 15]
```diff
@@ -0,0 +1,135 @@
...
+  &__show-all {
+    @apply shrink-0;
+  }
+
+  &__show-all-icon {
+    @apply mr-2;
+  }
+
+  &__sorting {
+    @apply z-10;
```
Comment (goldenmaya) [inline client-app/shared/catalog/components/category/category-horizontal-filters.vue:108]: VcDropdownMenu: use zIndex prop

File: client-app/shared/catalog/components/category/category-horizontal-filters.vue [hunk 16]
```diff
@@ -0,0 +1,135 @@
...
+  &__sorting-list {
+    @apply border-none;
+  }
+
+  &__sorting-item {
+    @apply pl-1 pr-5;
+  }
+
+  .vc-menu-item__inner {
+    @apply p-0;
```
Comment (goldenmaya) [inline client-app/shared/catalog/components/category/category-horizontal-filters.vue:128]: not good idea - fix parent paddings and remove this line, please.

File: client-app/shared/catalog/components/category/category-horizontal-filters.vue [hunk 17]
```diff
@@ -0,0 +1,135 @@
...
+  }
+
+  .vc-radio-button {
+    @apply py-2 ps-4 pe-8;
```
Comment (goldenmaya) [inline client-app/shared/catalog/components/category/category-horizontal-filters.vue:132]: fix after all changes

**Review by goldenmaya — CHANGES_REQUESTED** [review]

File: client-app/shared/catalog/components/category/filters-popup-sidebar.vue [hunk 18]
```diff
@@ -0,0 +1,164 @@
...
+
+const isExistSelectedPopupSidebarFacets = computedEager<boolean>(() =>
+  props.popupSidebarFilters.facets.some((facet) => facet.values.some((value) => value.selected)),
+);
+</script>
+
+<style lang="scss">
+.filters-popup-sidebar {
+  &--desktop {
+    @apply w-[358px] #{!important};
```
Comment (goldenmaya) [inline client-app/shared/catalog/components/category/filters-popup-sidebar.vue:141]: use media queries, not modifier

File: client-app/shared/catalog/components/facet-filter.vue [hunk 19]
```diff
@@ -109,19 +192,148 @@ const hasFade = computed(
...
+  &__item-input {
+    @apply w-full;
+  }
+
+  &__item-inner {
+    @apply flex items-center;
+  }
+
+  &--collapsable {
+    .facet-filter__content {
```
Comment (goldenmaya) [inline client-app/shared/catalog/components/facet-filter.vue:236]: No, use scss variable for modifier. Do not duplicate classes.
Example: https://github.com/VirtoCommerce/vc-theme-b2b-vue/blob/dev/client-app/ui-kit/components/organisms/push-message/vc-push-message.vue#L78

**Review by goldenmaya — CHANGES_REQUESTED** [review]

File: client-app/shared/catalog/components/category/category-horizontal-filters.vue [hunk 20]
```diff
@@ -0,0 +1,135 @@
...
+            </template>
+            {{ $t("common.buttons.sort_by") }}</VcButton
+          >
+        </template>
+        <template #content="{ close }">
+          <div class="py-2">
+            <VcMenuItem
+              v-for="sortingOption in PRODUCT_SORTING_LIST"
+              :key="sortingOption.id"
+              class="category-horizontal-filters__sorting-list"
```
Comment (goldenmaya) [inline client-app/shared/catalog/components/category/category-horizontal-filters.vue:43]: Why menu-items class=... __sorting-list?

File: client-app/shared/catalog/components/category/category-horizontal-filters.vue [hunk 21]
```diff
@@ -0,0 +1,135 @@
...
+              color="secondary"
+              size="sm"
+            >
+              <VcRadioButton
+                v-model="sortQueryParam"
+                size="sm"
+                :value="sortingOption.id"
+                :label="sortingOption.name"
+                class="category-horizontal-filters__sorting-item"
```
Comment (goldenmaya) [inline client-app/shared/catalog/components/category/category-horizontal-filters.vue:52]: Why ... __sorting-item?

**Review by goldenmaya — CHANGES_REQUESTED** [review]

**Review by goldenmaya — APPROVED** [review]

File: client-app/shared/catalog/components/category/category-horizontal-filters.vue (hunk 20 above)
Comment (ivan-kalachikov) [reply client-app/shared/catalog/components/category/category-horizontal-filters.vue:44]: changed name to __sorting-item and the next one to __sorting-input accordingly

File: client-app/shared/catalog/components/category/category-horizontal-filters.vue (hunk 21 above)
Comment (ivan-kalachikov) [reply client-app/shared/catalog/components/category/category-horizontal-filters.vue:53]: changed name to  __sorting-input

File: client-app/shared/catalog/components/facet-filter.vue [hunk 22]
```diff
@@ -106,22 +198,169 @@ const isAnchorAdded = computed(() => searchedValues.value.length > MAX_ITEMS_VIS
...
+
+      max-height: v-bind(maxHeight);
+    }
+  }
+
+  &__item-label {
+    @apply truncate;
+
+    #{$dropdown} & {
+      @apply mr-5 text-13 font-medium text-gray-500;
```
Comment (goldenmaya) [inline client-app/shared/catalog/components/facet-filter.vue:331]: text-13 - deprecated
font lato doesn’t support **font-medium**, **font-semibold**, **font-extrabold** (see font lato documentation)
text-gray-500 - we have to use colors from palette only

⬆️  fix it everywhere

**Review by goldenmaya — CHANGES_REQUESTED** [review]

**Review by goldenmaya — APPROVED** [review]

**Pull request merged**